lib/daru/vector.rb

```diff
@@ -199,19 +199,62 @@ def initialize source, opts={}
     #   # For vectors employing hierarchial multi index
     #
     def [](*input_indexes)
-      # Get a proper index object
-      indexes = @index[*input_indexes]
+      # Get array of positions indexes
+      positions = @index.pos(*input_indexes)
```

Comment:
Alternatively, think about it this way -
The user is  specifying positional indices to get elements AND their actual indexes. If the user wanted the hierarchically lower elements in a MultiIndex, she would just specify the top level index in `#[]`.

So I think that `v[0,1]` returning the whole MultiIndex and not a subset is a valid operation.